parser/digester.go

```diff
@@ -183,7 +183,7 @@ func (d *sqlDigester) normalize(sql string, keepHint bool) {
 		if tok == invalid {
 			break
 		}
-		if pos.Offset == len(sql) {
+		if pos.Offset == len(sql) || (pos.Offset == len(sql) - 1 && sql[pos.Offset] == ';') {
```

Comment:
Will changing delimiter at MySQL client affect it?

Reply:
Can the delimiter be changed by client?

Reply:
Ah, you are right, I'll test it.

Reply:
MySQL client will split and send the SQL to the server without a delimiter, so it won't have any impact. I saw it while capturing the packet.There is also a discussion here about the use of drivers, mentioning this. https://forums.mysql.com/read.php?39,297131,298822#msg-298822

Reply:
Oh I see the PR is to normalize the SQL query in a string literal, which is the argument of `tidb_encode_sql_digest`. My concern about a real SQL query is not related.